src/rpc/connection/crypto.c

```diff
@@ -446,6 +454,16 @@ int crypto_recv_initiate(struct crypto_context *cc, unsigned char *data)
 
   cc->state = TUNNEL_ESTABLISHED;
 
+  char *pluginkey_base16_encoded = MALLOC_ARRAY((PLUGINKEY_SIZE*2) + 1, char);
```

Comment:
cc->pluginkey should not be of type `string`. `string` type should only used for msgpack handling. just change `cc->pluginkey` to `char` array, like:

`char pluginkey[(PLUGINKEY_SIZE*2) + 1]`

and give the `base16_encode` function a pointer to this array.